version

```diff
@@ -8,19 +8,52 @@ MAJOR=1
 MINOR=9
 BRANCH_POINT=8228cea
 
+declare -a OTHERARGS
+help() {
+  cat <<-EOF
+Usage $0 [options]
+
+  --help This help
+  --ref  The reference for which version to output Defaults to HEAD
+
+Non-arg params
+
+- commit: output just the formatted commit hash
+- docker: output the version as used by the Docker tag
+EOF
+}
+
+while ! [ -z "$1" ]; do
+  arg="$1"
+  shift
+  case "$arg" in
+    --help)
+      help
+      exit 0
+      ;;
+    --ref)
+      REF="$1"
+      shift
+      ;;
+    *)
+      OTHERARGS+=("$arg")
+      ;;
+  esac
+done
+
 # Infer version
 # Number of commits since branch point
-COMMIT_NUMBER="$(git rev-list --count --first-parent $BRANCH_POINT..HEAD)"
-COMMIT_HASH=$(git rev-parse --short HEAD)
+COMMIT_NUMBER="$(git rev-list --count --first-parent $BRANCH_POINT..$REF)"
+COMMIT_HASH=$(git rev-parse --short $REF)
 
 # Echo commit hash
-if [ "$#" -eq 1 ] && [ "$1" == "commit" ]; then
+if [ "${OTHERARGS[0]}" == "commit" ]; then
```

Comment:
I get
```
› ./version commit
fatal: Needed a single revision
```
Where does `REF` default to `HEAD`?

Reply:
Ah, crap. I actually made this change on Marathon 1.8, and then forward ported it. In the process, I dropped the line where it default REF.